### ISSUE TITLE
Add Active Attributes to LWM2M client

### ISSUE THREAD

**Issue by arilwan** [eclipse/wakaama#411]
Using `wakaama` implementation of the `LWM2M` and `fiware IoTAgent`.

The scenario I want is `LWM2M -> IoTAgent -> Orion 
`
The `LWM2M` client registers with `IoTAgent`, But only with lazy attributes. How do I add active attributes for [temperature, humidity], so measures can be transmitted to `Orion`? Modify some files/code and rebuild client?

**Comment by sbertin-telular** [comment]
LWM2M doesn't have lazy or active attributes. What you probably want it to make some change to IoTAgent (I'm not familiar with it) to observe the desired client resources.

**Comment by arilwan** [comment]
I understand, my concern here is adding temperature and humidity as attributes.  What the `LWM2M` provision on registering with `IoTAgent` are mostly device specification attributes (model no, access control owner, etc) it does provision temp and humidity on registering with `IoTAgent`, please see below: 
```
$curl -X GET \
>   'http://localhost:4041/iot/devices' \
>   -H 'fiware-service: smartGondor' \
>   -H 'fiware-servicepath: /gardens'
{"count":1,"devices":[{"device_id":"raspiSensorTV","service":"smartGondor","service_path":"/gardens","entity_name":"raspiSensorTV:Device","entity_type":"Device","attributes":[],"lazy":[{"object_id":"Short Server ID","name":"Short Server ID","type":"Integer"},{"object_id":"Lifetime","name":"Lifetime","type":"Integer"},{"object_id":"Default Minimum Period","name":"Default Minimum Period","type":"Integer"},{"object_id":"Default Maximum Period","name":"Default Maximum Period","type":"Integer"},{"object_id":"Disable","name":"Disable","type":""},{"object_id":"Disable Timeout","name":"Disable Timeout","type":"Integer"},{"object_id":"Notification Storing When Disabled or Offline","name":"Notification Storing When Disabled or Offline","type":"Boolean"},{"object_id":"Binding","name":"Binding","type":"String"},{"object_id":"Registration Update Trigger","name":"Registration Update Trigger","type":""},{"object_id":"Object ID","name":"Object ID","type":"Integer"},{"object_id":"Object Instance ID","name":"Object Instance ID","type":"Integer"},{"object_id":"ACL","name":"ACL","type":"Integer"},{"object_id":"Access Control Owner","name":"Access Control Owner","type":"Integer"},{"object_id":"Manufacturer","name":"Manufacturer","type":"String"},{"object_id":"Model Number","name":"Model Number","type":"String"},{"object_id":"Serial Number","name":"Serial Number","type":"String"},{"object_id":"Firmware Version","name":"Firmware Version","type":"String"},{"object_id":"Reboot","name":"Reboot","type":""},{"object_id":"Factory Reset","name":"Factory Reset","type":""},{"object_id":"Available Power Sources","name":"Available Power Sources","type":"Integer"},{"object_id":"Power Source Voltage","name":"Power Source Voltage","type":"Integer"},{"object_id":"Power Source Current","name":"Power Source Current","type":"Integer"},{"object_id":"Battery Level","name":"Battery Level","type":"Integer"},{"object_id":"Memory Free","name":"Memory Free","type":"Integer"},{"object_id":"Error Code","name":"Error Code","type":"Integer"},{"object_id":"Reset Error Code","name":"Reset Error Code","type":""},{"object_id":"Current Time","name":"Current Time","type":"Time"},{"object_id":"UTC Offset","name":"UTC Offset","type":"String"},{"object_id":"Timezone","name":"Timezone","type":"String"},{"object_id":"Supported Binding and Modes","name":"Supported Binding and Modes","type":"String"},{"object_id":"Network Bearer","name":"Network Bearer","type":"Integer"},{"object_id":"Available Network Bearer","name":"Available Network Bearer","type":"Integer"},{"object_id":"Radio Signal Strength","name":"Radio Signal Strength","type":"Integer"},{"object_id":"Link Quality","name":"Link Quality","type":"Integer"},{"object_id":"IP Addresses","name":"IP Addresses","type":"String"},{"object_id":"Router IP Addresse","name":"Router IP Addresse","type":"String"},{"object_id":"Link Utilization","name":"Link Utilization","type":"Integer"},{"object_id":"APN","name":"APN","type":"String"},{"object_id":"Cell ID","name":"Cell ID","type":"Integer"},{"object_id":"SMNC","name":"SMNC","type":"Integer"},{"object_id":"SMCC","name":"SMCC","type":"Integer"},{"object_id":"Package","name":"Package","type":"Opaque"},{"object_id":"Package URI","name":"Package URI","type":"String"},{"object_id":"Update","name":"Update","type":""},{"object_id":"State","name":"State","type":"Integer"},{"object_id":"Update Supported Objects","name":"Update Supported Objects","type":"Boolean"},{"object_id":"Update Result","name":"Update Result","type":"Integer"},{"object_id":"Latitude","name":"Latitude","type":"String"},{"object_id":"Longitude","name":"Longitude","type":"String"},{"object_id":"Altitude","name":"Altitude","type":"String"},{"object_id":"Uncertainty","name":"Uncertainty","type":"String"},{"object_id":"Velocity","name":"Velocity","type":"Opaque"},{"object_id":"Timestamp","name":"Timestamp","type":"Time"},{"object_id":"SMS Tx Counter","name":"SMS Tx Counter","type":"Integer"},{"object_id":"SMS Rx Counter","name":"SMS Rx Counter","type":"Integer"},{"object_id":"Tx Data","name":"Tx Data","type":"Integer"},{"object_id":"Rx Data","name":"Rx Data","type":"Integer"},{"object_id":"Max Message Size","name":"Max Message Size","type":"Integer"},{"object_id":"Average Message Size","name":"Average Message Size","type":"Integer"},{"object_id":"StartOrReset","name":"StartOrReset","type":""}],"commands":[],"static_attributes":[]}]}
```

**Comment by sbertin-telular** [comment]
I think I understand. You would need to add LWM2M objects supporting the desired resources to the client. You probably want objects 3303 and 3304 from http://www.openmobilealliance.org/wp/OMNA/LwM2M/LwM2MRegistry.html.

**Comment by sbertin-telular** [comment]
You need to write the code to implement those objects on your device.

**Comment by dnav** [comment]
See examples/client/object_*.c files for examples on how to implement a LwM2M Object.

**Comment by arilwan** [comment]
@sbertin-telular @dnav 

So adding a `temperature variable `that should appear as attribute in the `iotagent `(and measurement observed) requires that bunch of code?

**Comment by dnav** [comment]
Maybe you are not familiar enough with the LightweightM2M protocol.

I suggest to watch these slides as an introduction: https://www.slideshare.net/zdshelby/oma-lightweightm2-mtutorial

Then this ones for more details: https://www.slideshare.net/OpenMobileAlliance/oma-lwm2m-tutorial-by-arm-to-ietf-ace

**Comment by arilwan** [comment]
Hi @dnav 
I created a temperature object `object_temperature.c`, and modified `lwm2mclient.c` and `lwm2mclient.h` as appropriate, but I am abit confused, here what I want to do.

I have already `LWM2M device` sending temperature measures to an `sqlite3` backend. Now I want the `object_temperatur.c` fetch recent temperature and forward to `LWM2M server` (actually within FIWARE).

I implemented the `db transaction` with `prv_temperature_read()` of the `object_temperature.c` but I'm not sure how the `prv_set_temperature()` function works. Should I pass the temperature value to this function to get reported to server?  Or just implement (do `stringToUri`) from within `prv_temperature_read()`.
